plugins/modem-manager/fu-mm-utils.c

```diff
@@ -56,8 +56,22 @@ fu_mm_utils_get_udev_port_info (GUdevDevice	 *device,
 
 	if (out_port_ifnum != NULL)
 		*out_port_ifnum = port_ifnum;
-	if (out_device_sysfs_path != NULL)
-		*out_device_sysfs_path = g_steal_pointer (&device_sysfs_path);
+	if (out_device_sysfs_path != NULL) {
+		/* if device sysfs path was set for a previous port,
+		 * make sure they are all equal */
+		if (*out_device_sysfs_path != NULL) {
+			if (g_strcmp0 (device_sysfs_path, *out_device_sysfs_path) != 0) {
```

Comment:
It does feel a bit dodgy using out_device_sysfs_path like this

Reply:
Maybe I could always return the device sysfs path as output, and do the check in the caller side?

Reply:
Ya, that works too.

Reply:
Done!